context-free.md

```diff
@@ -2,21 +2,30 @@
 
 ##  Who is this programming language for?
 
+Context Free is useful for users who wish to create math-based images with repetited geometric patterns with ease by writing human-readable instructions.  
 
 ## What is easy to do in this language? Why is it easy?
 
+In Context Free, it's easy to generate images with repetitive geometric patterns and the user can make the patterns be generated a few times spinning in a direction they specify. This is easy because Context Free lets the user define a "rule" and within the block where they are defining the rule, they can call the rule, making patterns repetitive. 
+
 
 ## What is hard to do in this language? Why is it hard?
 
+Because the coordinates and numbers for things like rotation are very mathematical, and scaled to the output screen in a way that's hard to intuitively grasp without a few trials and errors, it's hard to draw an image that requires a lot of small, complex details. In other words, it's hard for the user to imagine what the "rule" they are writing would end up in the output image. 
```

Comment:
In general, a DSL is supposed to make it easier to accomplish certain tasks for a domain expert. They don't necessarily accomplish those goals, but that is the intent.

CF seems like a language that would benefit from an IDE or workbench of some kind - what sort of features do you think are missing that would be well supplemented by tools, but don't merit being first-class citizens of the language?